sdk/synapse/Azure.Analytics.Synapse.Artifacts/tests/DataFlowClientLiveTests.cs

```diff
@@ -15,47 +18,79 @@ namespace Azure.Analytics.Synapse.Tests.Artifacts
-            {
-                DataFlowResource actualDataFlow = await DataFlowClient.GetDataFlowAsync(expectedDataFlow.Name);
-                Assert.AreEqual(expectedDataFlow.Name, actualDataFlow.Name);
-                Assert.AreEqual(expectedDataFlow.Id, actualDataFlow.Id);
-            }
+            DataFlowClient client = CreateClient();
+            await using DisposableDataFlow flow = await DisposableDataFlow.Create (client, this.Recording);
+
+            AsyncPageable<DataFlowResource> dataFlows = client.GetDataFlowsByWorkspaceAsync ();
+            Assert.GreaterOrEqual((await dataFlows.ToListAsync()).Count, 1);
+        }
+
+        [Test]
+        public async Task GetDataFlow()
+        {
+            DataFlowClient client = CreateClient();
+            await using DisposableDataFlow flow = await DisposableDataFlow.Create (client, this.Recording);
+
+            DataFlowResource dataFlow = await client.GetDataFlowAsync (flow.Name);
+            Assert.AreEqual (flow.Name, dataFlow.Name);
         }
 
         [Test]
-        public async Task TestCreateDataFlow()
+        public async Task RenameDataFlow()
         {
-            string dataFlowName = Recording.GenerateName("DataFlow");
-            DataFlowCreateOrUpdateDataFlowOperation operation = await DataFlowClient.StartCreateOrUpdateDataFlowAsync(dataFlowName, new DataFlowResource(new DataFlow()));
-            DataFlowResource dataFlow = await operation.WaitForCompletionAsync();
-            Assert.AreEqual(dataFlowName, dataFlow.Name);
+            DataFlowClient client = CreateClient();
+
+            // Non-disposable as we'll rename it underneath (and have to clean up ourselves)
+            DataFlowResource resource = await DisposableDataFlow.CreateResource (client, this.Recording);
+
+            string newFlowName = Recording.GenerateAssetName("DataFlow2");
+
+            DataFlowRenameDataFlowOperation renameOperation = await client.StartRenameDataFlowAsync (resource.Name, new ArtifactRenameRequest () { NewName = newFlowName } );
+            await renameOperation.WaitForCompletionAsync ();
+
+            DataFlowResource dataFlow = await client.GetDataFlowAsync (newFlowName);
+            Assert.AreEqual (newFlowName, dataFlow.Name);
+
+            DataFlowDeleteDataFlowOperation operation = await client.StartDeleteDataFlowAsync (newFlowName);
+            await operation.WaitForCompletionAsync ();
         }
 
         [Test]
-        public async Task TestDeleteDataFlow()
+        public async Task DeleteDataFlow()
         {
-            string dataFlowName = Recording.GenerateName("DataFlow");
+            DataFlowClient client = CreateClient();
 
-            DataFlowCreateOrUpdateDataFlowOperation createOperation = await DataFlowClient.StartCreateOrUpdateDataFlowAsync(dataFlowName, new DataFlowResource(new DataFlow()));
-            await createOperation.WaitForCompletionAsync();
+            // Non-disposable as we'll be deleting it ourselves
```

Comment:
I'm not sure what this comments means.   I think you're saying "This resource is disposable but we're not going to dispose it so that we can control deletion."  - am I close?

Reply:
It was "we're getting the raw resource and not using the Disposable for a reason here. If that wasn't clear, maybe I should just nuke the comment.